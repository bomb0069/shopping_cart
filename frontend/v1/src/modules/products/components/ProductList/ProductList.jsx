import React from 'react'
import Grid from '@material-ui/core/Grid'

import ProductCard from '../ProductCard'

const ProductList = ({
  productItems=[]
}) => {

  return (
    <Grid container spacing={5} style={{marginTop: '2em'}}>
      {productItems.map(item => (
        <Grid
          item
          xs={12}
          sm={4}
          key={'grid' + item.product_id}
        >
          <ProductCard
            key={item.product_id}
            {...item}
          />
        </Grid>
      ))}
    </Grid>
  )
}

export default ProductList