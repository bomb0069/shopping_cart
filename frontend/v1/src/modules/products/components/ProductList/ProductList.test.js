import React from 'react'
import { shallow } from 'enzyme'

import ProductList from './ProductList'
import ProductCard from '../ProductCard'

describe('<ProductList />', () => {
  let component

  beforeEach(() => {
    component = shallow(<ProductList />)

  })

  it('should render with no errors', () => {
    
    expect(component).toMatchSnapshot()
  })

  it('should render with product cards', () => {
    const productItems = [
      {
        id: 6,
        name: 'Princess Palace',
        price: 24.95,
        brand: 'Coolkidz',
        age: '3_to_5',
        availibility: 'InStock',
        gender: 'Female'
      },
      {
        id: 15,
        name: 'Sailboat',
        price: 24.95,
        brand: 'Coolkidz',
        age: 'over8',
        availibility: 'InStock',
        gender: 'Male'
      }    
    ]
    component.setProps({
      productItems
    })

    expect(component).toMatchSnapshot()
    expect(component.find(ProductCard).length).toEqual(productItems.length)
  })
})