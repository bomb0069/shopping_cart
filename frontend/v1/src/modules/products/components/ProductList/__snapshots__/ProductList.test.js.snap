// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> should render with no errors 1`] = `ShallowWrapper {}`;

exports[`<ProductList /> should render with product cards 1`] = `ShallowWrapper {}`;
